cirq/circuits/circuit.py

```diff
@@ -276,8 +276,9 @@ def __radd__(self, other):
         if not isinstance(other, (ops.Operation, Iterable)):
             return NotImplemented
         # Auto wrap OP_TREE inputs into a circuit.
-        result = Circuit(other)
-        return result.__iadd__(self)
+        result = self.copy()
+        result._moments[:0] = Circuit(other)._moments
```

Comment:
This doesn't appear to validate the new moment against the device.

Reply:
Fixed.

Reply:
Is validating the entire new circuit needed?  I would just validate the new moment:

```
result._moments[:0] = Circuit(other, device=self.device)._moments
```

Reply:
There are constraints on operations that can be applied in sequence on the same qubits for certain devices, so validating the whole circuit is necessary to catch that issue.

Reply:
OK, but Circuit.insert only validates the moment.  This is a separate issue from this PR.